roles/openshift_master/tasks/main.yml

```diff
@@ -36,48 +59,57 @@
   command: systemctl daemon-reload
   when: install_result | changed
 
-- name: Create certificate parent directory if it doesn't exist
-  file:
-    path: "{{ openshift_cert_parent_dir }}"
-    state: directory
-
 - name: Create config parent directory if it doesn't exist
   file:
-    path: "{{ openshift_master_config | dirname }}"
+    path: "{{ openshift_master_config_dir }}"
     state: directory
 
-# TODO: should probably use a template lookup for this
-# TODO: should allow for setting --etcd, --kubernetes options
-# TODO: recreate config if values change
-- name: Use enterprise default for oreg_url if not set
-  set_fact:
-    oreg_url: "openshift3_beta/ose-${component}:${version}"
-  when: openshift.common.deployment_type == 'enterprise' and oreg_url is not defined
+- name: Create the master certificates if they do not already exist
+  command: osadm create-master-certs --hostnames={{ openshift.common.hostname }},{{ openshift.common.public_hostname }} --master={{ openshift.master.api_url }} --public-master={{ openshift.master.public_api_url }} --cert-dir={{ openshift_master_config_dir }} --overwrite=false
+  args:
+    creates: "{{ openshift_master_config_dir }}/master.server.key"
+
+- name: Create the policy file if it does not already exist
+  command: osadm create-bootstrap-policy-file --filename={{ openshift_master_policy }}
+  args:
+    creates: "{{ openshift_master_policy }}"
+  notify:
+  - restart openshift-master
+
+- name: Create the scheduler config
+  template:
+    dest: "{{ openshift_master_scheduler_conf }}"
+    src: scheduler.json.j2
+  notify:
+  - restart openshift-master
 
-- name: Use online default for oreg_url if not set
-  set_fact:
-    oreg_url: "docker-registry.ops.rhcloud.com/openshift3_beta/ose-${component}:${version}"
-  when: openshift.common.deployment_type == 'online' and oreg_url is not defined
+- name: Create the htpasswd file if needed
+  copy:
+    dest: "{{ item.filename }}"
+    content: ""
+    mode: 0600
+    force: no
+  when: item.kind == 'HTPasswdPasswordIdentityProvider'
+  with_items: openshift.master.identity_providers
 
+# TODO: add the validate parameter when there is a validation command to run
 - name: Create master config
-  command: >
-    /usr/bin/openshift start master --write-config
-    --config={{ openshift_master_config }}
-    --portal-net={{ openshift.master.portal_net }}
-    --master={{ openshift.master.api_url }}
-    --public-master={{ openshift.master.public_api_url }}
-    --listen={{ 'https' if openshift.master.api_use_ssl else 'http' }}://0.0.0.0:{{ openshift.master.api_port }}
-    {{ ('--images=' ~ oreg_url) if (oreg_url | default('', true) != '') else '' }}
-    {{ ('--nodes=' ~ openshift_node_ips | join(',')) if (openshift_node_ips | default('', true) != '') else '' }}
-  args:
-    chdir: "{{ openshift_cert_parent_dir }}"
-    creates: "{{ openshift_master_config }}"
+  template:
+    dest: "{{ openshift_master_config_file }}"
+    src: master.yaml.v1.j2
```

Comment:
@jwhonce yes, however write config does not support all of the options available in the config file.  Using write config and then modifying the generated yaml file seemed a bigger risk than doing a full template. 

In discussions with @deads2k and @liggitt We will need to add some testing to verify that new keys added are flagged (By using --write-config and comparing the keys displayed against a generated template).  I haven't had a chance to work up a test for this yet, but it would be nice to have available as part of the automated testing (jenkins would probably be best, since we'd probably want to do regularly scheduled runs that would send alerts to major contributors). That way we can attempt to detect these earlier.  

I thought I saw an origin issue for tracking config related changes and automatically alerting people, but I can't seem to find it now (or it could be a sleep deprived delusion that I thought it existed).